src/main/java/com/google/codeu/data/Datastore.java

```
@@ -26,71 +26,225 @@
+                System.err.println(entity.toString());
+                e.printStackTrace();
+            }
+        }
+        return markers;
     }
 
-    return messages;
-
-  }
+    public void storeMarker(UserMarker marker) {
+        Entity markerEntity = new Entity("UserMarker");
+        markerEntity.setProperty("lat", marker.getLat());
+        markerEntity.setProperty("lng", marker.getLng());
+        markerEntity.setProperty("content", marker.getContent());
+        datastore.put(markerEntity);
+    }
+    
+    public List<Event> getEvents(){
+        List<Event> events = new ArrayList<>();
+    
+        Query query = new Query("Event");
+        PreparedQuery results = datastore.prepare(query);
+        
+        for (Entity entity : results.asIterable()) {
+            try {
+                
+                String idString = entity.getKey().getName();
+                UUID eventId = UUID.fromString(idString);
+                String speaker =  (String) entity.getProperty("speaker");
+                String organization =  (String) entity.getProperty("organization");
+                Date eventDate =  (Date) entity.getProperty("eventDate");
+                Location location =  (Location) entity.getProperty("location");
+                List<String> amenities =  getAmenities(eventId);
+                String externalLink =  (String) entity.getProperty("externalLink");
+                PublicType publicType =  (PublicType) entity.getProperty("publicType");
+                int ownerId =  (int) entity.getProperty("ownerId");
+                List<ThreadComment> thread =  getThread(eventId);
+                long timeStamp =  (long) entity.getProperty("timestamp");
+            
+                Event event = new Event(eventId, timeStamp, speaker, organization, eventDate, location, amenities, externalLink, publicType, ownerId);
+                event.copyThread(thread);
+                events.add(event);
+            } catch (Exception e) {
```

Usually it's good practice to only catch a specific set of Exception, and throw the rest

If there's a need to catch all exception, please add a comment here.

---

Added this line: //An exception can occur here for multiple reasons (Type casting error, any property not existing, key error, etc...)